### PR TITLE
[CIS-1512] Move tests under specific folders

### PR DESCRIPTION
**Pull request by bielikb**
### 🔗 Issue Link
CIS-1512

### 🎯 Goal

Move tests under shared folder in our repo

### 🛠 Implementation

Tests files are moved from the `Sources` folder under the respective subfolder in `Tests`. Tree folder for tests was replicated based on the `Sources` tree folder structure.

### 📝 Changes

* Tests, Mocks, Snapshots are now located under `Tests` folder.
* `Package.swift`, Podspecs no longer specifies the blacklist
* `GenerateSPMList` pre-commit hook is gone

Screenshot 2022-03-15 at 16 52 54

### 🧪 Testing

No need for testing

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎉 GIF

Reshuffling

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -38,236 +38,10 @@ let package = Package(
 // ** ⚠️ GENERATED, do not edit directly below this point **
 
 var streamChatSourcesExcluded: [String] { [
-    "Database/DataStore_Tests.swift",
-    "Database/DatabaseSession_Mock.swift",
-    "Database/DatabaseSession_Tests.swift",
-    "Database/DTOs/UserDTO_Tests.swift",
-    "Database/DTOs/ChannelMuteDTO_Tests.swift",
-    "Database/DTOs/ChannelDTO_Tests.swift",
-    "Database/DTOs/CurrentUserDTO_Tests.swift",
```
Comment (nuno-vieira) [inline Package.swift:47]: Don't we need to update the script that does this?

Comment (bielikb) [reply Package.swift:47]: Done. script removed!

File: Package.swift [hunk 2]
```diff
@@ -38,236 +38,10 @@ let package = Package(
...
-    "Controllers/MemberListController/MemberListController+SwiftUI_Tests.swift",
-    "ChatClient_Mock.swift"
 ] }
 
 var streamChatUIFilesExcluded: [String] { [
-    "ChatMessageList/TypingIndicatorView_Tests.swift",
-    "ChatMessageList/Reactions/ChatMessageReactionAuthorsVC/ChatMessageReactionAuthorsVC_Tests.swift",
     "ChatMessageList/Reactions/ChatMessageReactionAuthorsVC/__Snapshots__/ChatMessageReactionAuthorsVC_Tests/test_defaultAppearance_shouldNotRenderUnavailableReactions.default-light.png",
     "ChatMessageList/Reactions/ChatMessageReactionAuthorsVC/__Snapshots__/ChatMessageReactionAuthorsVC_Tests/test_defaultAppearance.default-light.png",
     "ChatMessageList/Reactions/ChatMessageReactionAuthorsVC/__Snapshots__/ChatMessageReactionAuthorsVC_Tests/test_defaultAppearance_whenOnlyOneReaction_shouldUseSingularLocalization.default-light.png",
```
[the remark this answers is not in the document]
Comment (bielikb) [reply Package.swift]: Done

File: StreamChat.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1943,20 +1956,6 @@
 			remoteGlobalIDString = 790881FC25432B7200896F03;
 			remoteInfo = StreamChatUI;
```
[the remark this answers is not in the document]
Comment (bielikb) [reply StreamChat.xcodeproj/project.pbxproj:1956]: I moved the snapshots under related subfolders in `Tests`.

File: Package.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (nuno-vieira) [reply Package.swift]: Hum 🤔 Don't we need to move them to the sources folder now? Would be interesting to test if SPM actually works. I'm afraid our CI job to test SPM Integration is not catching this. But these snapshot jpg are not part of the test target sources, so they should still be excluded

File: Package.swift (hunk 1 above)
Comment (nuno-vieira) [reply Package.swift:47]: I think the script still needs to exist, but we need now to exclude the snapshots in the test target 🤔

Comment (bielikb) [reply Package.swift:47]: The only reason the script was in place was to exclude the tests from the Swift Package.

Fortunately, with this change this is no longer needed. Also our Package manifest became more readable :)

Comment (nuno-vieira) [reply Package.swift:47]: Yes I just saw, that, nevermind, there are no Test targets in the Package.swift 👍

File: Package.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (bielikb) [reply Package.swift]: The SwiftPM integration works just fine. Our manifest points towards `Sources` folder.

Screenshot 2022-03-16 at 13 29 39

File: Package.swift (hunk 1 above)
Comment (nuno-vieira) [reply Package.swift:47]: I did find something tho 🤔 I was testing the SPM Integration project, and it seems the local repo doesn't get updated at all. So the SPM Integration job is actually not doing anything 😞 But ofc, this out of scope of the PR, but is something we need to keep in mind

Comment (bielikb) [reply Package.swift:47]: I can look on that part in one of the following PRs ;)

**Review by b-onc — APPROVED** [review]

**Pull request merged**